tests/unit/test_datafeeds.py

```diff
@@ -120,29 +192,81 @@ def test_upvest():
+        feed._raw_data = zoltu_json
+        assert feed.get_gas_price('slow') == Web3.toWei(41, 'gwei')
+        assert feed.get_gas_price('medium') == Web3.toWei(58, 'gwei')
+        assert feed.get_gas_price('fast') == Web3.toWei(67, 'gwei')
+        assert feed.get_gas_price('fastest') == Web3.toWei(70, 'gwei')
+        assert feed.get_gas_price() == feed.get_gas_price('fast')  # Default
+        parsed_gas_prices = feed.gas_prices
 
-    def mock_gas_strategy(web3, tx=None):
-        return mocked_gas_price
+    with patch('nucypher.utilities.datafeeds.ZoltuGasPriceDatafeed._parse_gas_prices', autospec=True):
+        ZoltuGasPriceDatafeed.gas_prices = dict()
+        with patch.dict(ZoltuGasPriceDatafeed.gas_prices, values=parsed_gas_prices):
+            gas_strategy = feed.construct_gas_strategy()
+            assert gas_strategy("web3", "tx") == Web3.toWei(67, 'gwei')
 
-    # In normal circumstances, the first datafeed (Etherchain) will return the gas price
-    with patch('nucypher.utilities.datafeeds.EtherchainGasPriceDatafeed.construct_gas_strategy',
-               return_value=mock_gas_strategy):
-        assert datafeed_fallback_gas_price_strategy("web3", "tx") == mocked_gas_price
 
-    # If the first datafeed in the chain fails, we resort to the second one
-    with patch('nucypher.utilities.datafeeds.EtherchainGasPriceDatafeed._probe_feed',
+def test_datafeed_median_gas_price_strategy():
+
+    mock_etherchain_gas_price = 1000
+    mock_upvest_gas_price = 2000
+    mock_zoltu_gas_price = 4000
+    mock_rpc_gas_price = 42
+
+    def construct_mock_gas_strategy(gas_price) -> Callable:
+        def _mock_gas_strategy(web3, tx=None):
+            return gas_price
+        return _mock_gas_strategy
+
+    # In normal circumstances, all datafeeds in the strategy work, and the median is returned
+    with patch('nucypher.utilities.datafeeds.UpvestGasPriceDatafeed.construct_gas_strategy',
+               return_value=construct_mock_gas_strategy(mock_upvest_gas_price)):
+        with patch('nucypher.utilities.datafeeds.EtherchainGasPriceDatafeed.construct_gas_strategy',
+                   return_value=construct_mock_gas_strategy(mock_etherchain_gas_price)):
+            with patch('nucypher.utilities.datafeeds.ZoltuGasPriceDatafeed.construct_gas_strategy',
+                       return_value=construct_mock_gas_strategy(mock_zoltu_gas_price)):
+                datafeed_median_gas_price_strategy = construct_datafeed_median_strategy()
+                assert datafeed_median_gas_price_strategy("web3", "tx") == median([mock_etherchain_gas_price,
+                                                                                   mock_upvest_gas_price,
+                                                                                   mock_zoltu_gas_price])
+
+    # If, for example, Upvest fails, the median is computed using the other two feeds
+    with patch('nucypher.utilities.datafeeds.UpvestGasPriceDatafeed._probe_feed',
```

Comment:
These mocks and assertions are very repetitive and can be reduced in a helperfunction, or possibly fixturization.  Additionally, the nested `with` blocks can be avoided entirely if you use the `mocker`, `module_mocker`, or `session_mocker` fixtures, since they have their own setup and tear down methods.

Reply:
Agreed, hard to read